bokehjs/src/coffee/embed.ts

```diff
@@ -88,8 +102,25 @@ function _init_comms(target: string, doc: Document): void {
     } catch (e) {
       logger.warn(`Jupyter comms failed to register. push_notebook() will not function. (exception reported: ${e})`)
     }
-  } else
-    console.warn('Jupyter notebooks comms not available. push_notebook() will not function');
+  } else if (doc.roots()[0].id in kernels) {
+    logger.info(`Registering JupyterLab comms for target ${target}`)
+    const kernel = kernels[doc.roots()[0].id]
+    for (const id in kernel._comms) {
```

Comment:
What does this loop do? This relies on a private attribute on ``kernel`` for in JupyterLab, so I'm hesitant to include it.

Reply:
I think it updates the onMsg handler on comms that have already been set up. It's not quite clear to me in what scenario the comm would already have been registered though.

Reply:
I see the same logic for classic notebooks (which presumably has a reason? I don't know how to git blame the previous embed.coffee), but I can't think of how it'd happen in a JLab context. I'd proposed deleting it.

Reply:
Looks like it was me 2 years in https://github.com/bokeh/bokeh/pull/5014 😞 

I'll have a go at removing it in classic notebooks and jupyterlab to see what happens.

Reply:
Here's what I said about it, I don't think any of this is true now, but I'd have to double check:

>Because now the template registers a comms target with Jupyter immediately, so it's not dependent on BokehJS being loaded. When bokeh does load it will first look if a comm has already been initialized for the specified target and attach itself with a callback using Comm.on_msg. Hope that makes sense. It feels a little bit dirty but ensures the comm target is always registered in time for the Comm to be created and initialized in Python.

Reply:
Okay I'm now fairly confident this is not needed anymore.

Reply:
Great - thanks for finding that out.

Reply:
> git blame the previous embed.coffee

Unfortunately due to how git handles renames and the fact that we use squash merges, this is far from being easy. You will have to find a commit or a tag  (`<rev>`) where rename happened and then use `git blame <rev>~1 -- file_name`.